engine/src/main/java/org/terasology/logic/behavior/tree/Task.java

```diff
@@ -91,6 +92,10 @@ public Status tick(float deltaSeconds) {
                     LOG.warn("update of Task {} returned invalid state {}", this.getClass(), newStatus);
                 }
                 status = newStatus;
+                if(newStatus == Status.FAILURE || newStatus == Status.SUCCESS) {
```

Comment:
Why is this better than calling `onTerminate()` in the corresponding switch cases?